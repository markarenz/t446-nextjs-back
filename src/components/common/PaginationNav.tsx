import Link from 'next/link';
import styles from '../../styles/modules/PaginationNav.module.scss';

type Props = {
  pageNum: number;
  itemsPerPage: number;
  itemsLoaded: number;
  numItems: number;
  path: string;
};

const PaginationNav: React.FC<Props> = ({
  pageNum,
  itemsPerPage,
  itemsLoaded,
  numItems,
  path
}) => {
  const firstItem = (pageNum - 1) * itemsPerPage + 1;
  const lastItem = firstItem + itemsLoaded - 1;
  const moreItems = lastItem + 1 <= numItems;
  return (
    <div className={styles.root} data-testid="pagination-nav">
      <span className={styles.navBtnWrapper}>
        {pageNum > 1 && (
          <Link href={`${path}${pageNum - 1}`} passHref>
            <span className="scaleHover">&#9664;</span>
          </Link>
        )}
      </span>
      <span className={styles.pageNumsWrapper}>
        {firstItem} - {lastItem}
      </span>
      <span className={styles.navBtnWrapper}>
        {moreItems && (
          <Link href={`${path}${pageNum + 1}`} className="scaleHover" passHref>
            <span className="scaleHover">&#9654;</span>
          </Link>
        )}
      </span>
    </div>
  );
};

export default PaginationNav;
