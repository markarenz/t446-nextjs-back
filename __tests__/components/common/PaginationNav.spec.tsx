import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import PaginationNav from '../../../src/components/common/PaginationNav';

const defaultProps = {
  pageNum: 1,
  itemsPerPage: 10,
  itemsLoaded: 10,
  numItems: 12,
  path: '/test/'
};
const altProps = {
  ...defaultProps,
  pageNum: 2,
  itemsLoaded: 2
};

describe('PaginationNav', () => {
  it('renders component', async () => {
    render(<PaginationNav {...defaultProps} />);
    const element = screen.queryByTestId('pagination-nav');
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });
  it('renders component- last page', async () => {
    render(<PaginationNav {...altProps} />);
    const element = screen.queryByTestId('pagination-nav');
    expect(element).toBeInTheDocument();
  });
});
