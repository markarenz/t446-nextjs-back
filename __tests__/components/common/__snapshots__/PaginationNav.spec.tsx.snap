// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNav renders component 1`] = `
<div
  class="root"
  data-testid="pagination-nav"
>
  <span
    class="navBtnWrapper"
  />
  <span
    class="pageNumsWrapper"
  >
    1
     - 
    10
  </span>
  <span
    class="navBtnWrapper"
  >
    <span
      class="scaleHover"
      href="/test/2"
    >
      â–¶
    </span>
  </span>
</div>
`;
